operator/internal/central/extensions/reconcile_tls_test.go

```diff
@@ -581,11 +587,186 @@ func Test_checkCertRenewal(t *testing.T) {
 	}
 }
 
+func Test_checkCertificateTimeValidity(t *testing.T) {
+	cases := map[string]struct {
+		now             string
+		notBefore       string
+		notAfter        string
+		wantErrContains string
+	}{
+		"should return error if validity range is invalid": {
+			now:             "2026-01-01T00:00:00Z",
+			notBefore:       "2030-01-01T00:00:00Z",
+			notAfter:        "2025-01-01T00:00:00Z",
+			wantErrContains: "certificate expires at",
+		},
+		"should return error if certificate is not yet valid": {
+			now:             "2024-01-01T00:00:00Z",
+			notBefore:       "2025-01-01T00:00:00Z",
+			notAfter:        "2030-01-01T00:00:00Z",
+			wantErrContains: "certificate lifetime start",
+		},
+		"should return error if certificate is expired": {
+			now:             "2031-01-01T00:00:00Z",
+			notBefore:       "2025-01-01T00:00:00Z",
+			notAfter:        "2030-01-01T00:00:00Z",
+			wantErrContains: "certificate expired",
+		},
+		"should return no error for valid certificate": {
+			now:       "2026-06-01T00:00:00Z",
+			notBefore: "2025-01-01T00:00:00Z",
+			notAfter:  "2030-01-01T00:00:00Z",
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			now, err := time.Parse(time.RFC3339, c.now)
+			require.NoError(t, err)
+
+			cert := generateTestCertWithValidity(t, c.notBefore, c.notAfter)
+			r := &createCentralTLSExtensionRun{currentTime: now}
+
+			err = r.checkCertificateTimeValidity(cert)
+
+			if c.wantErrContains != "" {
+				require.Error(t, err)
+				assert.ErrorContains(t, err, c.wantErrContains)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGenerateCentralTLSData_Rotation(t *testing.T) {
```

Comment:
These are basically the same checks as in the other package, right? Could we share them somehow? 🤔

Reply:
I'm also bothered by this but I'm not sure what to do. They are almost the same checks (it's a bit more extensive as it checks the leaf cert too).

I could simply just remove `TestGenerateCentralTLSData_Rotation` as it doesn't bring anything new, it's there just to make the coverage tool happy.